### ISSUE TITLE
Infinite hangs in Tor Browser when trying to decrypt a UTC/JSON wallet file in the dist release

### ISSUE THREAD

**Issue by leafcutterant** [kvhnuke/etherwallet#570]
MyEtherWallet CX 3.8.9
Tor Browser 7.0 (based on Mozilla Firefox 52.1.2) (32-bit)

When using the dist release in the Tor Browser and trying to unlock an encrypted UTC/JSON wallet file, after entering the password and pressing the Unlock button, the tab hangs up and stays unresponsive for an indefinite time. It does this both when I enter a correct and an incorrect password.

This might be a browser performance limitation as opening a wallet via unencrypted private key in the same setting works, and the wallet decryption works well when I open dist in the most up-to-date non-Tor Browser version of Firefox, Firefox portable 53.0.3 (64-bit).

**Comment by tayvano** [comment]
Thank you for letting us know.

Decrypting these keystore files is quite intensive and we try to make improvements. I do know that on Firefox previously it would time out; pressing "continue..." after the first timeout would usually result in it unlocking. 

Do you have the ability to see a console in Tor with any errors that may  be occurring?

We can address errors but the decrypt times are part of the encryption mechanism so that one cannot bruteforce as quickly / easily.

**Comment by leafcutterant** [comment]
Of course, you have to include an appropriate number of rounds to ensure security. I just thought that there is something in Tor Browser that is different and could be addressed. The FF version in plain Firefox and Tor Browser is pretty close (53 and 52), do you think it could be a bug in 52?

By the way, I'm not sure of my memory, but I *think* I did successful UTC/JSON decryptions in Tor Browser in the past, but those were different versions of MyEtherWallet dist and Tor Browser. I can't remember what versions those were.

This is my full browser console output minus the reflows, from opening a new tab to hanging. Nothing very telling I think:

```Warning: ‘nsIOService::NewChannel()’ deprecated, please use ‘nsIOService::NewChannel2()’  HTTPSRules.js:253:18
SSL Observatory WARN: Failed to initialize NSS component:Error: couldn't find function symbol in library
Expected color but found ‘context-fill’.  Error in parsing value for ‘fill’.  Declaration dropped.  svg.css:24:9
Error in parsing value for ‘fill-opacity’.  Declaration dropped.  svg.css:25:17
Expected color but found ‘context-stroke’.  Error in parsing value for ‘stroke’.  Declaration dropped.  svg.css:26:11
Error in parsing value for ‘stroke-opacity’.  Declaration dropped.  svg.css:27:19
Error in parsing value for ‘stroke-width’.  Declaration dropped.  svg.css:28:17
Error in parsing value for ‘stroke-dasharray’.  Declaration dropped.  svg.css:29:21
Error in parsing value for ‘stroke-dashoffset’.  Declaration dropped.  svg.css:30:22
Tor NOTICE: Bootstrapped 90%: Establishing a Tor circuit 
Tor NOTICE: Tor has successfully opened a circuit. Looks like client functionality is working. 
Tor NOTICE: Bootstrapped 100%: Done 
[06-11 18:57:37] Torbutton NOTE: Initializing security-prefs.js
[06-11 18:57:37] Torbutton NOTE: security-prefs.js initialization complete
Tor NOTICE: New control connection opened from 127.0.0.1. 
[06-11 18:57:37] Torbutton WARN: no SOCKS credentials found for current document.
Key event not available on some keyboard layouts: key=“r” modifiers=“accel,alt” id=“toggleReaderMode”  browser.xul
Key event not available on GTK2: key=“u” modifiers=“accel shift” id=“torbutton-new-identity-key”  browser.xul
Key event not available on some keyboard layouts: key=“i” modifiers=“accel,alt,shift” id=“key_browserToolbox”  browser.xul
Use of nsIFile in content process is deprecated.  NetUtil.jsm:304:12
Use of nsIFile in content process is deprecated.  filesystem.js:40:9
GET 
XHR 
<sensitive data removed> [HTTP/1.1 301 Moved Permanently 4061ms]
POST 
http://ocsp.digicert.com/ [HTTP/1.1 200 OK 1548ms]
GET 
XHR 
<sensitive data removed> [HTTP/1.1 200 OK 2106ms]
POST 
http://ocsp.digicert.com/ [HTTP/1.1 200 OK 580ms]
[06-11 18:58:09] Torbutton WARN: no SOCKS credentials found for current document.
Error in parsing value for ‘-webkit-text-size-adjust’.  Declaration dropped.  etherwallet-master.min.css:2:155
Unknown pseudo-class or pseudo-element ‘-webkit-inner-spin-button’.  Ruleset ignored due to bad selector.  etherwallet-master.min.css:2:1425
Unknown pseudo-class or pseudo-element ‘-webkit-search-cancel-button’.  Ruleset ignored due to bad selector.  etherwallet-master.min.css:2:1600
Unknown property ‘orphans’.  Declaration dropped.  etherwallet-master.min.css:3:588
Unknown property ‘widows’.  Declaration dropped.  etherwallet-master.min.css:3:597
Error in parsing value for ‘color’.  Declaration dropped.  etherwallet-master.min.css:3:1079
Error in parsing value for ‘font-size’.  Declaration dropped.  etherwallet-master.min.css:3:1824
Error in parsing value for ‘font-size’.  Declaration dropped.  etherwallet-master.min.css:3:1994
Error in parsing value for ‘margin-bottom’.  Declaration dropped.  etherwallet-master.min.css:3:3942
Error in parsing value for ‘padding-bottom’.  Declaration dropped.  etherwallet-master.min.css:3:5635
Error in parsing value for ‘margin’.  Declaration dropped.  etherwallet-master.min.css:3:5649
Error in parsing value for ‘margin-bottom’.  Declaration dropped.  etherwallet-master.min.css:3:6023
Expected end of value but found ‘1.4’.  Error in parsing value for ‘padding’.  Declaration dropped.  etherwallet-master.min.css:3:6435
Expected end of value but found ‘1.4’.  Error in parsing value for ‘margin’.  Declaration dropped.  etherwallet-master.min.css:3:6450
Error in parsing value for ‘margin-bottom’.  Declaration dropped.  etherwallet-master.min.css:3:7402
Error in parsing value for ‘padding’.  Declaration dropped.  etherwallet-master.min.css:3:7783
Expected end of value but found ‘0.7’.  Error in parsing value for ‘margin’.  Declaration dropped.  etherwallet-master.min.css:3:7797
Error in parsing value for ‘margin-bottom’.  Declaration dropped.  etherwallet-master.min.css:3:16227
Error in parsing value for ‘margin-bottom’.  Declaration dropped.  etherwallet-master.min.css:3:21173
Error in parsing value for ‘margin-bottom’.  Declaration dropped.  etherwallet-master.min.css:3:22786
Unknown pseudo-class or pseudo-element ‘-ms-input-placeholder’.  Ruleset ignored due to bad selector.  etherwallet-master.min.css:3:24125
Unknown pseudo-class or pseudo-element ‘-webkit-input-placeholder’.  Ruleset ignored due to bad selector.  etherwallet-master.min.css:3:24176
Expected media feature name but found ‘-webkit-min-device-pixel-ratio’.  etherwallet-master.min.css:3:24601
Error in parsing value for ‘min-height’.  Declaration dropped.  etherwallet-master.min.css:3:25503
Unknown property ‘user-select’.  Declaration dropped.  etherwallet-master.min.css:3:28961
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped.  etherwallet-master.min.css:3:29483
Unknown property ‘user-select’.  Declaration dropped.  etherwallet-master.min.css:3:39880
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped.  etherwallet-master.min.css:3:40602
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped.  etherwallet-master.min.css:3:42081
Expected ‘none’, URL, or filter function but found ‘progid’.  Error in parsing value for ‘filter’.  Declaration dropped.  etherwallet-master.min.css:3:43695
Error in parsing value for ‘box-shadow’.  Declaration dropped.  etherwallet-master.min.css:3:48971
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped.  etherwallet-master.min.css:3:57877
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped.  etherwallet-master.min.css:3:57986
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped.  etherwallet-master.min.css:3:59200
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped.  etherwallet-master.min.css:3:59254
Unknown property ‘user-select’.  Declaration dropped.  etherwallet-master.min.css:3:68365
Unknown property ‘min-widht’.  Declaration dropped.  etherwallet-master.min.css:3:68773
Error in parsing value for ‘display’.  Declaration dropped.  etherwallet-master.min.css:3:74779
Error in parsing value for ‘display’.  Declaration dropped.  etherwallet-master.min.css:3:74889
Error in parsing value for ‘display’.  Declaration dropped.  etherwallet-master.min.css:3:80977
Error in parsing value for ‘display’.  Declaration dropped.  etherwallet-master.min.css:3:88769
Error in parsing value for ‘display’.  Declaration dropped.  etherwallet-master.min.css:3:91959
Error in parsing value for ‘display’.  Declaration dropped.  etherwallet-master.min.css:3:92347
Error in parsing value for ‘display’.  Declaration dropped.  etherwallet-master.min.css:3:92735
OPTIONS 
XHR 
https://api.myetherapi.com/eth [HTTP/1.1 200 OK 2167ms]
file:///C:/MyEtherWallet/images/fav/favicon.ico
POST 
XHR 
https://api.myetherapi.com/eth [HTTP/1.1 200 OK 719ms]```

**Comment by tayvano** [comment]
@leafcutterant Following up on this -- was this a keystore generated on MyEtherWallet or via Mist/geth?

**Comment by leafcutterant** [comment]
Gotcha! It was generated with geth (the latest release at the time of posting the issue)—however, I tried it and dist (in Tor Browser) has no problem opening a UTC/JSON  wallet file which was generated with MEW CX (in Chromium).

It could be that the specs for encrypting the wallet file are different in geth and CX and Tor Browser can't cope with it. FYI, I also tested the geth  wallet file on CX (Chromium), and i imported perfectly.
